crypto/evp/pmeth_check.c

```diff
@@ -17,23 +17,43 @@
 #include "crypto/evp.h"
 #include "evp_local.h"
 
+/*
+ * Returns:
+ *  1   True
+ *  0   False
+ * -1   Unsupported (use legacy path)
+ */
+static int try_provided_check(EVP_PKEY_CTX *ctx, int selection)
+{
+    EVP_KEYMGMT *keymgmt;
+    void *keydata;
+
+    if (evp_pkey_ctx_is_legacy(ctx))
+        return -1;
+
+    keymgmt = ctx->keymgmt;
+    keydata = evp_pkey_export_to_provider(ctx->pkey, ctx->libctx,
+                                          &keymgmt, ctx->propquery);
```

Comment:
Does this keydata need to be stored in the EVP_PKEY to ensure it is later freed? Or has that already been done somewhere (I couldn't figure out where if so)?

Reply:
No need, that's already stored in `ctx->pkey`